okhttp/src/main/java/okhttp3/internal/http/BridgeInterceptor.java

```diff
@@ -81,9 +83,12 @@ public BridgeInterceptor(CookieJar cookieJar) {
       requestBuilder.header("Accept-Encoding", "gzip");
     }
 
-    List<Cookie> cookies = cookieJar.loadForRequest(userRequest.url());
-    if (!cookies.isEmpty()) {
-      requestBuilder.header("Cookie", cookieHeader(cookies));
+    List<Cookie> cookieJarCookies = cookieJar.loadForRequest(userRequest.url());
+    if (!cookieJarCookies.isEmpty()) {
+      // Concatenate cookies from the cookie jar and the user's request. If a cookie is in both the
+      // request and the cookie jar we send both! Well-behaved calling code shouldn't do this.
```

Comment:
What specifically means that well-behaved code shouldn't do this?  Curious, because it wasn't immediately clear to me that concatenating instead of taking the request cookies *over* cookie-jar cookies was the right approach.

Reply:
I was thinking that if you’re manually providing cookies, it’s your responsibility to make sure those cookies don’t conflict with the cookie store.

But you’re right, I think it’s a bit of a bad use-case, and gets us into solving problems we don’t have enough information to solve. Might be simpler to keep our current policy, where if you have a `CookieJar` that’s where the cookies all come from.

@nicktrav can you help us here? Some context on how you ended up having two different things trying to supply cookies?

Reply:
I would have assumed the opposite, that putting cookies on the request was intentional to override the default behaviour.  But deferring to you.